src/containers/App.js

```diff
@@ -0,0 +1,101 @@
+
+
+// src/containers/App.js
+
+import React, { Component } from 'react';
+import { ALL_TODOS, ACTIVE_TODOS, COMPLETED_TODOS } from '../constants';
+
+import Header from '../components/Header';
+import Todos from '../components/Todos';
+import Footer from '../components/Footer';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+		
+		this.state = {
+			show: ALL_TODOS,
+			items: [
+				{ label: 'item 1', done: 0 },
+				{ label: 'item 2', done: 0 },
+				{ label: 'item 3', done: 0 },
+				{ label: 'item 4', done: 0 },
+			],
+		};
+	}
+	
+	setFilter(show) {
+		this.setState({ show: show });
+	}
+	
+	addEntry(label) {
+		// I was about to add code to stop it adding duplicate labels (since we use the labels as keys)
+		// but I tested and found it doesn't add them
+		// this has to be because we're using the label as the key

```

Comment:
Ahh yes, line 37 stops duplicates.  Not sure what comment you are referring to...